src/udev/dmi_memory_id/dmi_memory_id.c

```diff
@@ -444,21 +439,12 @@ static void dmi_decode(const struct dmi_header *h) {
                         break;
 
                 log_debug("Use: System Memory");
-                printf("MEMORY_ARRAY_LOCATION=%s\n",
-                       dmi_memory_array_location(data[0x04]));
-                if (data[0x06] != 0x03) {
-                        printf("MEMORY_ARRAY_EC_TYPE=%s\n",
-                               dmi_memory_array_ec_type(data[0x06]));
-                }
-                if (DWORD(data + 0x07) == 0x80000000) {
-                        if (h->length >= 0x17)
-                                dmi_print_memory_size("MEMORY_ARRAY", "MAX_CAPACITY", -1, QWORD(data + 0x0F), MEMORY_SIZE_UNIT_BYTES);
-                } else {
-                        uint64_t capacity;
-
-                        capacity = DWORD(data + 0x07);
-                        dmi_print_memory_size("MEMORY_ARRAY", "MAX_CAPACITY", -1, capacity, MEMORY_SIZE_UNIT_KB);
-                }
+                dmi_memory_array_location(data[0x04]);
+                dmi_memory_array_ec_type(data[0x06]);
+                if (DWORD(data + 0x07) != 0x80000000)
```

Comment:
This doesn't look like the correct simplification of those conditionals.

Reply:
I unset the flag since we probably need to clarify that before merging.

Reply:
Sorry, I cannot find any error on this change. Could you elaborate more?

Reply:
`data[0x06] == 0x03` will now print MEMORY_ARRAY_EC_TYPE=None, but before it didn't.

Reply:
I made `dmi_memory_array_ec_type()`  not show `None`. But in the above comment, @hadess said 'Why?' So, now the EC_TYPE= is always shown.

Reply:
> I made `dmi_memory_array_ec_type()` not show `None`. But in the above comment, @hadess said 'Why?' So, now the EC_TYPE= is always shown.

Not sure why you're commenting about the EC Type changes, when the problem is the conditional the line below.

Reply:
But I didn't see the fact that the conditional got reversed. That's why I said that I couldn't review the monster diff of patches :/